### DISCUSSION TITLE
Adding "Generic" board definitions.

### DISCUSSION THREAD

**Discussion by bdurbrow** [FPGAwars/icestudio#838] [New Boards Requests]
I would like to add (or have added for me, if that's easier for you to do) a couple of "generic" board definitions. These would consist of just a fairly basic serial programmer interface; and mappings for all the pins on the QFP 100 and QFP 144 package FPGAs. The idea is that you could select the appropriate generic board definition; and use IceStudio to develop for a custom board containing that FPGA; without the IceStudio project having to integrate support for each and every one-off or custom board that somebody wants to make.

I think that I would like to create & contribute a lightweight protocol for the serial download; a little bit more sophisticated than the BlackIce interface (so that you could send other commands to the MCU than just the bitstream for the FPGA); but still keep it lightweight. I would like to implement this as an Arduino library for the MCU side; and a simple C program for the PC side.

I believe I have enough documentation to both create schematics for these generic boards; and the programmer firmware/library/sketch.

What I'm a little fuzzy on is what I need to do facilitate getting it integrated into IceStudio. What do I need to provide; and where should I look for examples of how the other boards are supported?

**Comment by cavearr** [comment]
Hello! I am reorganizing the issues, trying to give more visibility and be able to better organize and prioritize the work and my roadmap (I will publish it very soon), I am converting some issues that are **new feature requests** as **ideas** on Github **discussions**.

Thanks for understanding.
